apps/cow-fi/components/SearchBar.tsx

```diff
@@ -173,6 +242,36 @@ export const SearchBar: React.FC<SearchBarProps> = ({ articles }) => {
   const [isFocused, setIsFocused] = useState(false)
   const [highlightedIndex, setHighlightedIndex] = useState(-1)
   const inputRef = useRef<HTMLInputElement>(null)
+  const searchContainerRef = useRef<HTMLDivElement>(null)
+  const phrasesCache = useRef<Record<string, string[]>>({})
+  const MAX_CACHE_ENTRIES = 100 // Limit cache size
+  const isMediumUp = !useMediaQuery(Media.upToMedium(false))
+
+  // Helper to manage cache size
+  const addToCache = (key: string, value: string[]) => {
+    // If cache is full, remove oldest entry
+    const cacheKeys = Object.keys(phrasesCache.current)
+    if (cacheKeys.length >= MAX_CACHE_ENTRIES) {
+      delete phrasesCache.current[cacheKeys[0]]
```

Comment:
You delete only the first key. But what of cacheKeys.length is 150? Then I think we should delete the first 50 keys

Reply:
Also outdated